devices/Bmm150/README.md

```diff
@@ -0,0 +1,181 @@
+    public abstract void ReadBytes(I2cDevice i2CDevice, Register reg, Span<byte> readBytes);
+}
+```
+
+For example the I2C basic implementation is the following:
+
+```csharp
+public class Bmm150I2c : Bmm150I2cBase
+{
+    public override byte ReadByte(I2cDevice i2cDevice, Register reg)
+    {
+        i2cDevice.WriteByte((byte)reg);
+        return i2cDevice.ReadByte();
+    }
+
+    public override void ReadBytes(I2cDevice i2cDevice, Register reg, Span<byte> readBytes)
+    {
+        i2cDevice.WriteByte((byte)reg);
+        i2cDevice.Read(readBytes);
+    }
+
+    public override void WriteRegister(I2cDevice i2cDevice, Register reg, byte data)
+    {
+        Span<byte> dataout = stackalloc byte[] { (byte)reg, data };
+        i2cDevice.Write(dataout);
+    }
+}
+```
+
+If you have to use a different I2C interface, you have to use the constructor where you can pass it:
+
+```csharp
+Bmm150 = new Bmm150(_i2cDevice, new Bmm150Attached(), false);
+```
+
+## Circuit
+
+Only I2C is supported in this version.
+
+* SCL - SCL
+* SDA - SDA
+* VCC - 3.3V
+* GND - GND
+
+Depending on the version you have, you may have to select I2C over SPI. This is done in different way depending on the board you'll have. 
+
+## Not supported/implemented features of the Bmm150
+
+* Device Self-Tests
+* Device Reset
+* Toggle operation modes (defaults to normal mode)
+
+
+## Testing
+
+Unit tests project is in \Bmm150.tests. You can use VS2019 built-in test capabilites as follows:
+
+![unit tests](./vs2019_unit_tests.png)
+
+```csharp
+[TestMethod]
+public void TestCompensateVector3()
+{
+    uint rhall = 42;
+    Vector3 rawMagnetormeterData = new Vector3 { X = 13.91375923, Y = -28.74289894, Z = 10.16711997 };
+    Bmm150TrimRegisterData trimRegisterData = new Bmm150TrimRegisterData()
+    {
+        dig_x1 = 0,
+        dig_x2 = 26,
+        dig_xy1 = 29,
+        dig_xy2 = -3,
+        dig_xyz1 = 7053,
+        dig_y1 = 0,
+        dig_y2 = 26,
+        dig_z1 = 24747,
+        dig_z2 = 763,
+        dig_z3 = 0,
+        dig_z4 = 0
+    };
+
+    double x = Bmm150Compensation.Compensate_x(rawMagnetormeterData.X, rhall, trimRegisterData);
+    double y = Bmm150Compensation.Compensate_y(rawMagnetormeterData.Y, rhall, trimRegisterData);
+    double z = Bmm150Compensation.Compensate_z(rawMagnetormeterData.Z, rhall, trimRegisterData);
+
+    // Calculated value should be: -1549.91882323
+    Assert.Equal(Math.Ceiling(x), Math.Ceiling(-1549.918823), "Unexpected x-axis value.");
+
+    // Calculated value should be: 3201.80615234
+    Assert.Equal(Math.Ceiling(y), Math.Ceiling(3201.80615234), "Unexpected y-axis value.");
+
+    // Calculated value should be: 26.20077896
+    Assert.Equal(Math.Ceiling(z), Math.Ceiling(26.20077896), "Unexpected z-axis value.");
+}
+```
+
+## Reference
+
+Documentation for the Bmm150 can be found here: https://www.bosch-sensortec.com/media/boschsensortec/downloads/datasheets/bst-bmm150-ds001.pdf
```

Comment:
please sur the pattern for links with [ ] and ( )

Reply:
solved